### DISCUSSION TITLE
Using this for a non-website servers?

### DISCUSSION THREAD

**Discussion by Vel-San** [nginx-proxy/nginx-proxy#1776] [Q&A] [answered]
Hey there! I'm facing some troubles when hosting a steam-dedicated server using 2 containers (e.g. 2 servers of 1 game).

My docker run for first server works perfectly fine, using:

>docker run -it --name Server_1 -p 0.0.0.0:7707:7707/udp -p 0.0.0.0:7708:7708/udp -p 0.0.0.0:7717:7717/udp -p 0.0.0.0:28852:28852/udp -p 0.0.0.0:28852:28852/tcp -p 0.0.0.0:8075:8075/tcp -p 0.0.0.0:20560:20560/udp -p 0.0.0.0:20560:20560/tcp -v /home/vel/kf-stuff/server-volume:/home/steam/servers/kf -e STEAM_USER=USER -e STEAM_PASSWORD='PASS' -e STEAM_CODE=$1 --env-file=/home/vel/kf-stuff/env_file vel7an/kf1-docker

Now, obviously when trying to run a 2nd container using:

>docker run -it --name Server_2 -p 0.0.0.0:7709:7709/udp -p 0.0.0.0:7708:7708/udp -p 0.0.0.0:7719:7719/udp -p 0.0.0.0:28852:28852/udp -p 0.0.0.0:28852:28852/tcp -p 0.0.0.0:8077:8077/tcp -p 0.0.0.0:20560:20560/udp -p 0.0.0.0:20560:20560/tcp -v /home/vel/kf-stuff/server-volume-2:/home/steam/servers/kf -e STEAM_USER=USER -e STEAM_PASSWORD='PASS' -e STEAM_CODE=$1 --env-file=/home/vel/kf-stuff/env_file_2 vel7an/kf1-docker

Ports **7709, 7719 & 8077** can be changed just fine, but the other Steam/Server ports **7708, 28852 & 20560** cannot be changed in the server itself, so I'm assuming these need to be the same through all containers, and this is where nginx-proxy can come in handy.

My question is, do you know wether nginx-proxy can work for this use-case? I tried using it but ended up failing (Used it on a single server/container and the server didn't show up at all).

If you would like to experiment or at least view the project itself, here are some links:

- [Repo](https://github.com/Vel-San/killing-floor-docker)
- [DockerHub IMG](https://hub.docker.com/r/vel7an/kf1-docker)

Thanks in advance ^_^

**Comment by junderw** [comment, marked as answer]
You would basically have to create your own custom nginx.tmpl file... but I think it would be doable.

It's not going to be documented here, you would need to dig into the weeds of how nginx conf files work, how the template engine works and then use that knowledge to make your own nginx.tmpl file.

**Reply by Vel-San** [reply]
Thanks for the response. I was a dummy and was able to fix the issue.

I tested with --net=host and both servers worked, then I saw the the 2nd server isn't actually `Visible` in the server browser but the first one was, so figured it was a port-forwarding issue. Turns out it was OK to change even the Steam ports.

Thanks regardless :)!

**Reply by Vel-San** [reply]
I am unable to lose the discussion so I will simply mark your comment as answer.
